hubspot3/test/test_tickets.py

```diff
@@ -49,3 +57,207 @@ def test_get_all_tickets():
+                'removedAssociations': [],
+                'newListMemberships': [],
+                'removedListMemberships': [],
+            }
+        }
+    ]
+    response_body_batch = {
+        '47005994': {
+            'objectType': 'TICKET',
+            'portalId': 5282301,
+            'objectId': 47005994,
+            'properties': {
+                'hs_lastcontacted': {
+                    'versions': [{
+                        'name': 'hs_lastcontacted',
+                        'value': '1571427728000',
+                        'timestamp': 1571411169418,
+                        'sourceId': 'TicketsRollupProperties',
+                        'source': 'CALCULATED',
+                        'sourceVid': []
+                    }, {
+                        'name': 'hs_lastcontacted',
+                        'value': '1571415325000',
+                        'timestamp': 1571410899877,
+                        'sourceId': 'TicketsRollupProperties',
+                        'source': 'CALCULATED',
+                        'sourceVid': []
+                    }],
+                    'value': '1571427728000',
+                    'timestamp': 1571411169418,
+                    'source': 'CALCULATED',
+                    'sourceId': 'TicketsRollupProperties'
+                },
+                'hs_last_email_activity': {
+                    'versions': [{
+                        'name': 'hs_last_email_activity',
+                        'value': 'SENT_TO_CONTACT',
+                        'timestamp': 1571411169118,
+                        'sourceId': 'TicketsRollupProperties',
+                        'source': 'CALCULATED',
+                        'sourceVid': []
+                    }, {
+                        'name': 'hs_last_email_activity',
+                        'value': 'REPLY_FROM_CONTACT',
+                        'timestamp': 1571409899877,
+                        'sourceId': 'TicketsRollupProperties',
+                        'source': 'CALCULATED',
+                        'sourceVid': []
+                    }],
+                    'value': 'SENT_TO_CONTACT',
+                    'timestamp': 1571411169118,
+                    'source': 'CALCULATED',
+                    'sourceId': 'TicketsRollupProperties'
+                }
+            }
+        }
+    }
+    # TODO should add more complex data to test, this is not testing the full complexity of the method
+    ids = [47005994]
+    properties = ['hs_lastcontacted', 'hs_last_email_activity']
+    params_batch = {"propertiesWithHistory": p for p in properties}
+    params_recent = {
+        'objectId': 47005994,
+        'timestamp': 1571411169000,
+        'changeType': changeType
+    }
+
+    mock_connection.set_responses([(200, json.dumps(response_body_recent)),
+                                   (200, json.dumps(response_body_batch)),
+                                   (200, '[]')])
+    if changeType == 'CHANGED':
```

Comment:
Here you're repeating the constant, which could be a problem if only of them is changed. One option is to `if` over the method, and other is to extract the constant to a variable.

```python
if changeType == 'CHANGED':
   changes_request_method = tickets_client.get_recently_modifier
else:
   changes_request_method = tickets_client.get_recently_created

changes = changes_request_method(time_offset=1571847002
```
or
```python
SAMPLE_OFFSET = 1571847002
if ....
```

In any case, `changeType` is not the standard snake_case. Is this common in the existing codebase?

Should tests even have this kind of logic?

Reply:
it was not standard, it is just that the tests are very similar, with just that parameter changing. I liked the first option, going to change to that